Core/AppURLs+StatisticsDependent.swift

```diff
@@ -0,0 +1,148 @@
+//
+//  AppURLs+StatisticsDependent.swift
+//  DuckDuckGo
+//
+//  Copyright © 2023 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import BrowserServicesKit
+
+public extension URL {
```

Comment:
Still looking at this but something that jumps out is the factory method names.  The [API Design Guidelines](https://www.swift.org/documentation/api-design-guidelines/) says:

> The first argument to initializer and [factory methods](https://en.wikipedia.org/wiki/Factory_method_pattern) calls should not form a phrase starting with the base name 

And since these aren't a value preserving operation the first parameter should not be omitted. 

So for instance:

```static func make(forPixelWithName pixelName: String, formFactor: String? = nil, includeATB: Bool = true) -> URL```

should probably be:

```static func makePixelURL(name: String, formFactor: String? = nil, includeATB: Bool = true) -> URL```

I think the reasoning here is that factory methods are effectively constructors and you wouldn't do that there either. 

@bwaresiak any thoughts on that?  We've had similar conversations in the past :) 

(Note, moved to a comment from main discussion)

Reply:
changed, looks much better!